field_search_panel/models/search_panel_mixin.py

```diff
@@ -0,0 +1,152 @@
+from lxml import etree
+
+from odoo import _, api, models
+from odoo.osv.expression import AND
+
+
+class SearchpanelMixin(models.AbstractModel):
+    _name = "search.panel.mixin"
+    _description = "Mixin Search Panel"
+
+    @api.model
+    def get_views(self, views, options=None):
+        result = super().get_views(views, options=options)
+        if (
+            "views" in result
+            and "search" in result["views"]
+            and "arch" in result["views"]["search"]
+        ):
+            result["views"]["search"]["arch"] = self._set_dynamic_search_panel(
+                result["views"]["search"]["arch"]
+            )
+        return result
+
+    @api.model
+    def _set_dynamic_search_panel(self, arch):
+        param = self.env["ir.config_parameter"].sudo()
+        searchpanel_field_ids = param.get_param("field_selection.searchpanel_field_ids")
+        if not searchpanel_field_ids:
+            return arch
+        searchpanel_field_ids = searchpanel_field_ids[1:-1]
+        searchpanel_field_ids = [int(x) for x in searchpanel_field_ids.split(",")]
+
+        fields = self.env["ir.model.fields"].search(
+            [
+                ("id", "in", searchpanel_field_ids),
+                ("model", "=", self._name),
+            ]
+        )
+
+        if not fields:
+            return arch
+
+        eview = etree.fromstring(arch)
+
+        search_panels = eview.xpath("//searchpanel")
+
+        if len(search_panels) == 0:
+            search_panel = etree.Element("searchpanel")
+        else:
+            search_panel = search_panels[0]
+
+        for field in fields:
+            efield = etree.Element("field", name=field.name)
+            search_panel.insert(0, efield)
+
+        eview.insert(0, search_panel)
+
+        return etree.tostring(eview, pretty_print=True)
+
+    @api.model
+    def web_search_read(
+        self,
+        domain=None,
+        fields=None,
+        offset=0,
+        limit=None,
+        order=None,
+        count_limit=None,
+    ):
+        """
+        Due to the search panel implementation, we can't use False as a boolean value.
+        So we use "True" and "False" as string values and we convert them to boolean.
+        """
+        new_domain = []
+        for leaf in domain:
+            if len(leaf) < 3:
+                new_domain.append(leaf)
+                continue
+            field, operator, value = leaf
+            field = self._fields[field]
```

Comment:
A domain could be in the format `parent_id.a_field`. That'll raise an exception in your case